serverlessworkflow/modules/ROOT/pages/eventing/working-with-callbacks.adoc

```diff
@@ -1,26 +1,43 @@
-Working with Callbacks
-======================
+= Callback state in Serverless Workflow
+
 :compat-mode!:
 // Metadata:
 :description: Working with callbacks
 :keywords: kogito, workflow, serverless, callback, event
+// links:
+:java_install: https://www.java.com/en/download/help/download_options.html
+:maven_install: https://maven.apache.org/install.html
+:docker_install: https://docs.docker.com/engine/install/
+:podman_install: https://docs.podman.io/en/latest/
+:kubectl_install: https://kubernetes.io/docs/tasks/tools/install-kubectl
+:java_install_url: https://www.java.com/en/download/help/download_options.html
+:maven_install_url: https://maven.apache.org/install.html
+:docker_install_url: https://docs.docker.com/engine/install/
+:podman_install_url: https://docs.podman.io/en/latest/
+:kubectl_install_url: https://kubernetes.io/docs/tasks/tools/install-kubectl
+:kn_cli_install_url: https://github.com/knative/client/blob/main/docs/README.md#installing-kn
+:kafka_doc_url: https://kafka.apache.org/documentation/ 
+
+This document describes the link:{spec_doc_url}#Callback-State[Callback state] and related examples. The Callback state peforms an action and waits for an event to produce the result of the action to resume the workflow. The action peformed by a Callback state is an asynchronous external service invocation. Therefore, the Callback state is suitable to perform  _fire&wait-for-result_ operations.
 
+From a Serverless Workflow perspective, asynchronous service indicates that the control is returned to the caller immediately without waiting for the action to be completed. Once the action is completed, a link:{cloud_events_url}[CloudEvent] is published to resume the Serverless Workflow.
 
-This guide discuss the link:{spec_doc_url}#Callback-State[Callback State]. This is a state that performs an action and waits for an event produced as a result of that action to resume the workflow. So it is suitable to perform __fire&wait-for-result__ operations. The action will normally be an external service invocation. This service is expected to be async.
+For the Serverless Workflow to identify the published CloudEvent it is waiting for, the external service developer includes the process instance ID in the CloudEvent header or use the xref:event-correlation-with-workflows[Serverless Workflow correlation mechanism]. 
 
-From the flow perspective, async means that the control will be returned to the caller almost immediately, without waiting for the actual operation to be finished. Once it ends, a link:{cloud_events_url}[Cloud Event] will be published to resume the flow.
+The Serverless Workflow correlation mechanism document focuses on the former mechanism that is based on the fact that each Serverless Workflow instance contains a unique identifier, which is generated automatically.
```

Comment:
This completely change the meaning. It is actually the other way around. This guide focus on the unique identifier approach, while correlation focus on a different approach base on filter attributes

Reply:
@fjtirado Can you please elaborate the meaning change in the following sentences:

**Original paragraph:** For the flow to know this published event is the one it is waiting for, the external service developer might include the Kogito process instance id in the event header or, preferably, use the xref:event-correlation-with-workflows[Serverless Workflow correlation mechanism]. This guide will focus on the former mechanism, which is based on the fact that every **Kogito Workflow instance has an automatically generated and unique identifier.**

**Modified paragraph:** For the Serverless Workflow to identify the published CloudEvent it is waiting for, the external service developer includes the process instance ID in the CloudEvent header or uses the xref:event-correlation-with-workflows[Serverless Workflow correlation mechanism]. 

The Serverless Workflow correlation mechanism document focuses on the former mechanism that is based on the fact that each **Serverless Workflow instance contains a unique identifier, which is generated automatically.**

Both paragraphs state that the workflow instance contains an identifier that is unique and automatically generated.

Reply:
Done